### DISCUSSION TITLE
Struggling with chapters, please help!

### DISCUSSION THREAD

**Discussion by stuartrosscraig** [audiamus/AaxAudioConverter#139] [Q&A]
Hi!

I'm struggling to understand how to input the chapters, please help!

Some of my chapters are way off what they're supposed to be, how can I edit them? I read the manual entries on chapters but I couldn't understand them exactly. I buy my audiobooks from audible and I'm converting them to m4b to use in other players, but only about half the chapter lengths come out correctly, so I need to manually change the others.

Thank you for your time :D

**Comment by audiamus** [comment]
I assume you have enabled "named chapters". It takes its input from `content_meta_<ASIN>.json`. You can edit that file, it's text and self explaining. But you may need to reformat it first to make it legible. And I'd recommend an audio editor like Audacity to find the correct chapter marks.

**Reply by stuartrosscraig** [reply]
This is great information! Thanks for replying - I'm new to this :D

I assume you're referring to the "Use named chapters, if available" checkbox in the Basic Settings? Yes, I've selected that :D 

Unfortunately, I don't think audible includes the chapter names in the downloadable files though - I spoke with them and they said I'd only see chapter numbers if I played the books in third party apps (I want to use Apple Books).

I download my audible audiobooks as single .aax files - do I need to go into those to find the "content_meta_<ASIN>.json"? How do I open them to find it?

The audible app already shows me the time markers for all the chapters, so if I can get to the "content_meta_<ASIN>.json", then I can just enter the times manually. And can I enter the chapter names manually too, while I'm there?

Thanks!

**Reply by stuartrosscraig** [reply]
Ok!  I've been reading more and I now understand the "content_meta_NAME.json" is a separate file -  I have one now (with all the correct info in it) but I can't make AaxAudioConverter use it!

I read the manual again and it looks like AaxAudioConverter should automatically use it if they're both in the same folder - but I have the aax and json files in a folder together and it won't work for me - it won't take the chapter names or the correct chapter markers...

Is there something I can do to explicitly point AaxAudioConverter at my json file?

Thanks so much :D

**Reply by stuartrosscraig** [reply]
It worked!! :D This is great :D

Previously I was using the audible files I downloaded from my audible library in-browser - and that doesn't work for me,

But when I use the audible files which I downloaded from the audible app, it does work! AaxAudioConverter uses the json file!

With a little trial and error, I now see it's all about the name of the aax file: it has to have "_XXXXXXXXXX" in it, with a code matching the name of the json file.

Perfect! Thanks so much for everything :D

**Reply by audiamus** [reply]
Good to hear. 

This XXXXXXXX is the "ASIN", a unique product ID assigned by Amazon.
 
You are aware that the Audible Win10 app is going to be discontinued soon? After that you can use [Book Lib Connect](https://github.com/audiamus/BookLibConnect) to download your books, which will also produce the `content_meta_ASIN.json` files.

**Reply by stuartrosscraig** [reply]
Aw! I didn't know that, thank you :D I've now downloaded Book Lib Connect.

Do you know if there a program that will make a json file for the aax files I've already downloaded (that I don't have json files for)?

Thanks :D

**Reply by audiamus** [reply]
The chapter data is separate and additional information that must be downloaded from the Audible web server. The only reasonable ways to obtain this are to re-download the books either with the Audible app (while still usable) or with Book Lib Connect.

**Reply by stuartrosscraig** [reply]
Thanks for everything :D You've really helped me :D
